azure-proto-core/ArmClient.cs

```diff
@@ -1,161 +1,109 @@
+    /// credential required.
+    /// TODO: What is appropriate naming for ArmClient , given that we would not liek to make distinctions between data and management.
     /// </summary>
     public class ArmClient
     {
         internal static readonly string DefaultUri = "https://management.azure.com";
 
-        public ArmClient()
-            : this(new Uri(DefaultUri), new DefaultAzureCredential(), null, new ArmClientOptions())
-        {
-        }
+        public Dictionary<string, string> ApiVersionOverrides { get; private set; }
 
-        public ArmClient(string defaultSubscriptionId)
-            : this(new Uri(DefaultUri), new DefaultAzureCredential(), defaultSubscriptionId, new ArmClientOptions())
-        {
-        }
+        public ArmClient(ArmClientOptions options = null)
+            : this(new Uri(DefaultUri), new DefaultAzureCredential(), null, options ?? new ArmClientOptions()) { }
 
-        public ArmClient(TokenCredential credential, string defaultSubscriptionId)
-            : this(new Uri(DefaultUri), credential, defaultSubscriptionId, new ArmClientOptions())
-        {
-        }
+        public ArmClient(string defaultSubscriptionId, ArmClientOptions options = null)
+            : this(new Uri(DefaultUri), new DefaultAzureCredential(), defaultSubscriptionId, options ?? new ArmClientOptions()) { }
 
-        public ArmClient(Uri baseUri, TokenCredential credential)
-            : this(baseUri, credential, null, new ArmClientOptions())
-        {
-        }
+        public ArmClient(TokenCredential credential, string defaultSubscriptionId, ArmClientOptions options = null)
+            : this(new Uri(DefaultUri), credential, defaultSubscriptionId, options ?? new ArmClientOptions()) { }
+
+        public ArmClient(Uri baseUri, TokenCredential credential, ArmClientOptions options = null)
+            : this(baseUri, credential, null, options ?? new ArmClientOptions()) { }
 
-        public ArmClient(
-            Uri baseUri,
-            TokenCredential credential,
-            string defaultSubscriptionId,
-            ArmClientOptions options)
+        public ArmClientOptions ClientOptions { get; private set; }
+
+        public ArmClient(Uri baseUri, TokenCredential credential, string defaultSubscriptionId, ArmClientOptions options)
         {
+            ClientOptions = options;
             ClientContext = new ArmClientContext(baseUri, credential);
             defaultSubscriptionId ??= GetDefaultSubscription().ConfigureAwait(false).GetAwaiter().GetResult();
-            DefaultSubscription = new SubscriptionOperations(
-                ClientContext,
-                new ResourceIdentifier($"/subscriptions/{defaultSubscriptionId}"));
+            DefaultSubscription = new SubscriptionOperations(ClientContext, new ResourceIdentifier($"/subscriptions/{defaultSubscriptionId}"), options);
```

Comment:
>new ResourceIdentifier [](start = 76, length = 22)

@m-nash, I feel there is an opportunity here to create some helper method for more structured ResourceIdentifier construction. There are still too much free typing going on here. Lets add a TODO for fit and finish.